actor/src/main/scala-2.12/org/apache/pekko/util/ByteString.scala

```diff
@@ -433,6 +447,20 @@ object ByteString {
       }
     }
 
+    override def indexOf(elem: Byte): Int = indexOf(elem, 0)
+    override def indexOf(elem: Byte, from: Int): Int = {
+      if (from >= length) -1
+      else {
+        var found = -1
+        var i = math.max(from, 0)
+        while (i < length && found == -1) {
+          if (bytes(startIndex + i) == elem) found = i
+          i += 1
+        }
+        found
```

Comment:
how about we extract the `startIndex + 1` to the start of the loop and returning a ` found - startIndex`？